### PR TITLE
Fix empty object crash bug

### PR DESCRIPTION
**Pull request by danielgtaylor**
This does two things:
1. Fix a crash when an element is of type `object` and there is no `content`, either in the base or any referenced type.
2. Modify the referencing examples to test a little bit more:
   - Test `object1` -> `object2` -> `object3` inheritance relationship to confirm that deep referencing does indeed work as expected.
   - Test dereferencing when passed-in refract is just a reference and nothing else, i.e. all information is really in the data structures section like is common in API Blueprint.

Example with empty objects (in Attributes Kit):
screen shot 2016-02-16 at 10 41 48

cc @Baggz

### CODE REVIEW

File: src/dereference.coffee [hunk 1]
```diff
@@ -20,7 +20,7 @@ module.exports = dereference = (root, dataStructures) ->
       # as a form of inheritance. We go through each property and transclude
       # as necessary, then output object element with the modified list of
       # members.
-      properties = root.content.slice(0)
+      properties = if root.content then root.content.slice(0) else []
```
Comment (Baggz) [inline src/dereference.coffee:23]: :+1:

**Comment by Baggz** [comment]
Looks good to me!

**Comment by Baggz** [comment]
@danielgtaylor 

May I ask you to add me to the [eidolon](https://www.npmjs.com/package/eidolon) NPM package as a collaborator? I would be able to push new versions.

**Comment by danielgtaylor** [comment]
@Baggz added you (and a few others) as collaborators on npm. Sorry about that!

**Comment by Baggz** [comment]
Thanks, @danielgtaylor!

**Pull request merged**